src/Features/LanguageServer/Protocol/Handler/Diagnostics/Public/PublicDocumentNonLocalDiagnosticSourceProvider.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics.DiagnosticSources;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics.Public;
+
+[ExportDiagnosticSourceProvider, Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class PublicDocumentNonLocalDiagnosticSourceProvider(
+    [Import] IGlobalOptionService globalOptions,
+    [Import] IDiagnosticAnalyzerService diagnosticAnalyzerService)
+    : AbstractDocumentDiagnosticSourceProvider(NonLocal)
+{
+    public const string NonLocal = "NonLocal_B69807DB-28FB-4846-884A-1152E54C8B62";
```

Comment:
@dibarbet I am still confused about what NonLocal source/provider is for. DO we need it?

Reply:
i am as well.  can this be answered?